### PR TITLE
Anguyen/argo spike

### PR DESCRIPTION
**Pull request by nguyena2**
# Type of PR

- [x] Documentation changes
- [ ] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [ ] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR
Argo-CD spike to test advanced features like Application-sets, Application-generators, sync-waves/hooks, and progressive syncs.

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/1183

### CODE REVIEW

**Comment by nguyena2** [comment]
@microsoft-github-policy-service agree [company="Microsoft"]

**Comment by nguyena2** [comment]
@microsoft-github-policy-service agree

File: spikes/argo-cd/manifests/cluster-generator/application-set-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        path: spikes/argo-cd/deploy/argo-cluster
+      destination:
+        server: '{{server}}'
+        namespace: ngsa
+      syncPolicy:
+        syncOptions:
+        - CreateNamespace=true
+        automated:
+          selfHeal: true
+          prune: true
```
Comment (joaquinrz) [inline spikes/argo-cd/manifests/cluster-generator/application-set-cluster.yaml:30]: add newline at end of file

File: spikes/argo-cd/manifests/git-generator/argo-cluster/config.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    "app": {
+        "source": "https://github.com/retaildevcrews/ngsa-asb",
+        "revision": "anguyen/argo-spike",
+        "path": "spikes/argo-cd/deploy/argo-cluster"
+    },
+    "cluster": {
+        "name": "argo-cluster",
+        "address": "https://kubernetes.default.svc"
+    }
+}
```
Comment (joaquinrz) [inline spikes/argo-cd/manifests/git-generator/argo-cluster/config.json:12]: add newline at end of file

File: spikes/argo-cd/manifests/git-generator/argo-cluster2/config.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    "app": {
+        "source": "https://github.com/retaildevcrews/ngsa-asb",
+        "revision": "anguyen/argo-spike",
+        "path": "spikes/argo-cd/deploy/argo-cluster2"
+    },
+    "cluster": {
+        "name": "argo-cluster2",
+        "address": "https://argocluste-argoresource-ce2cc3-w2pcxy1e.hcp.southcentralus.azmk8s.io:443"
+    }
+}
```
Comment (joaquinrz) [inline spikes/argo-cd/manifests/git-generator/argo-cluster2/config.json:12]: add newline at end of file

**Review by joaquinrz — CHANGES_REQUESTED** [review]
looks good, just add newline at the end of the files mentioned

File: spikes/argo-cd/manifests/git-generator/argo-cluster/config.json (hunk 2 above)
Comment (nguyena2) [reply spikes/argo-cd/manifests/git-generator/argo-cluster/config.json]: resolved

File: spikes/argo-cd/manifests/cluster-generator/application-set-cluster.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
```
Comment (joaquinrz) [inline spikes/argo-cd/manifests/cluster-generator/application-set-cluster.yaml:31]: one more new line missing :)

**Review by joaquinrz — APPROVED** [review]
Approved with comments

**Pull request merged**